src/dex/solidly/forks-override/spiritSwapV2.ts

```diff
@@ -0,0 +1,60 @@
+import { Solidly } from '../solidly';
```

Comment:
This file may be a mistake. Is it possible to incorporate fee changes into one instance of `Solidly` which is initialized with different parameters for different versions?

Because we have inheritance: `SpiritSwapV2 -> Solidly -> UniswapV2 -> SimpleExchange`. It is becoming very long :(

Or maybe @ColonelJ  has other opinion

Reply:
I kinda mimicked what we did for uniswap-v2 forks that supports dynamic fees https://github.com/paraswap/paraswap-dex-lib/blob/feat/solidly-forks/src/dex/uniswap-v2/excalibur.ts#L45. I honestly prefer this design as it ends up with less spaghetti code

Reply:
Though thinking of your variant idea. I can be generic implementations like SoldilyWithFeesOnFactory and SoldilyWithFeesOnPair